chef/site-cookbooks/wca/templates/gen-authorized-keys.sh.erb

```diff
@@ -16,8 +16,13 @@ fi
   --keyfile /home/cubing/.ssh/authorized_keys
 
 # Add keys used by Travis for deployment.
+# Note that we restrict Travis to running our deploy.sh script.
+# We use a trick from
+# http://larstobi.blogspot.com/2011/01/restrict-ssh-access-to-one-command-but.html
+# to pass command line arguments down to deploy.sh.
 authorized_keys_path=/home/cubing/.ssh/authorized_keys
 wca_documents_key_url="https://raw.githubusercontent.com/thewca/wca-documents/master/.travis/deploy_rsa.pub"
 echo "" >> $authorized_keys_path
 echo "# Keys for WCA Documents - Travis" >> $authorized_keys_path
-curl -s $wca_documents_key_url >> $authorized_keys_path
+travis_public_key=$(curl -s $wca_documents_key_url)
+echo 'command="/home/cubing/worldcubeassociation.org/scripts/deploy.sh ${SSH_ORIGINAL_COMMAND#* }",' $travis_public_key >> $authorized_keys_path
```

Comment:
Including the explanation for this string manipulation, which may not be obvious on first read:

> Notice the '#* ' right after SSH_ORIGINAL_COMMAND. The # modifier in bash is used to remove the smallest prefix pattern, and so '#* ' will remove everything up until and including the first space.

So we indeed strip the command we want to execute and keep only the positional arguments.

The blog post doesn't include that comma after the `command="xxx"` options, does it work too?

Reply:
May be a matter of preference, but we could specify the command line argument directly. This wouldn't require any *tricks* and would certainly do the job just fine!